### ISSUE TITLE
Automatically update fields in the database

### ISSUE THREAD

**Issue by Ant13731** [JacquesCarette/Drasil#2775]
Building off of #2675 and #2774, it might be nice to include all "complex" (or higher-level) chunks in their reduced forms for other fields of the chunk database. For example, everything that is in the `ConceptMap` field of a `ChunkDB` can be included under the `TermMap`, since the `ConceptChunks` used in a `ConceptMap` contains `IdeaDicts` which can be used in the `TermMap`. This reduces the need for a user to include the same chunk in both fields.

**Comment by Ant13731** [comment]
When trying to include all chunks that are a part of the `NamedIdea` class as an `IdeaDict`, some examples have chunks that use the same UID with different terms. For example, in _, running `make` with the filler for `IdeaDicts` results in some unwanted, stable-breaking changes in GamePhysics and SSP. In particular, GamePhysics' overuse of the `accj` UID is really prominent in both the log files (`debug/GamePhysics_SRS`) and stable. I don't think we want this, so I'll list the terms that need fixing here:

| UID | Example | Issues |
| --- | --- | --- |
| `accj` | GamePhysics | the term used for the concepts and instance models conflicts with the one used for `QuantityDict` |
| `mobShr` and `mobilizedShear` | SSP | General definition term for `mobShr` conflicts with the concept version. The underlying issue for this is that there should not be manual pluralization of the term since it is a nounphrase. |

**Comment by Ant13731** [comment]
Looking at the GamePhysics issue in a little more depth, I found for the IMod `rotMot` the following description of the variables:
![image](https://user-images.githubusercontent.com/69334555/129200301-b6838e5b-7a60-44cb-8c61-26bd3b9e5575.png)
Should the a_j variable description be replaced with "a_j is the acceleration of the jth body"? The term is already attached, just overwritten as of now.

**Comment by smiths** [comment]
Good catch @Ant13731.  You are correct that alpha_j is not the force.  It is the angular acceleration.  We should be clear that it is related to the rotational motion, not the translational motion.  Can you change it to "alpha_j is the angular acceleration of the j-th body (rad/s^2)".  Interestingly, the units were correct in the original; it is just the definition that is wrong.

**Comment by Ant13731** [comment]
#2788 should be completed before uncommenting the functions that link between `ChunkDB` fields

**Comment by JacquesCarette** [comment]
The basic idea is very good - the chunk hierarchy should indeed be used, instead of forcing the work onto the 'user'.

I'm not so happy with the implementation scheme. I would prefer to teach Drasil about its own hierarchy, and thus have the lookup functions do the work automatically.

That still leaves the issue of using the same UID for multiple things. In any single example, that should not be allowed at all.

> This kind of problem stems from some constructors that override the term from a given chunk while inheriting the same UID from that chunk.

That does indeed seem bad. I think I remember why: we're specializing from something generic to something more specific. Having a more specific term is good -- re-using the same UID is where things go wrong.

**Comment by balacij** [comment]
We've now gone $180^\circ$ on the chunk nesting. So, while the initial issue is closable, I just want to make sure that everything mentioned in the discussion is addressed somehow by other issues.

**Comment by JacquesCarette** [comment]
> We've now gone 180 ∘ on the chunk nesting. 

Yes we have!
